### DISCUSSION TITLE
Domotise n’importe quel compteur edf pour 20€ en 5 minutes - Haade.fr

### DISCUSSION THREAD

**Discussion by giscus** [haade-administrator/haade.fr#20] [Comments]
# Domotise n’importe quel compteur edf pour 20€ en 5 minutes - Haade.fr

Relève en 5 minutes la téléinfo dans home assistant sur n’importe quel compteur edf (ancien et linky) avec une clé micro téléinfo v3.0 par charles Hallard à 20€, par l’intermédiaire des modules mqtt et téléinfo2mqtt

https://haade.fr/fr/blog/domotise-linky-micro-teleinfo-homeassistant

**Comment by Goj34** [comment]
Bonjour, et merci pour ce tuto au top. 
J'ai un soucis après l'installation. J'ai bien le capteur avec les MAJ créé  mais aucun autre capteur ne remonte dans MQTT. autre chose (peut être en lien) les leds de mon dongle ne s'allument pas comme les tiennes. 
défaut du dongle USB ?  un avis ?  merci

**Reply by haade-administrator** [reply]
salut reconnait-il correctement ttyACMX, as-tu bien paramétré le broker mosquitto, as-tu essayé de le brancher directement sur une prise usb pour voir l'activation des leds ?

**Reply by haade-administrator** [reply]
si tout a été testé peut être un défaut de la clé, voir avec Charles

**Reply by Goj34** [reply]
teleinfo2mqtt
12:08:26.979  INFO teleinfo2mqtt:
    Starting teleinfo2mqtt with configuration = {
      emitInterval: 10,
      hassDiscovery: true,
      hassDiscoveryPrefix: 'homeassistant',
      logLevel: 'debug',
      mqttBaseTopic: 'teleinfo',
      mqttUrl: 'mqtt://core-mosquitto',
      mqttUser: 'homeassistant',
      mqttPassword: '<hidden>',
      mqttTlsClientKey: undefined,
      mqttTlsClientCert: undefined,
      mqttTlsCaChain: undefined,
      mqttTlsRejectUnauthorized: true,
      serial: '/dev/ttyACM0',
      ticMode: 'history'
    }
12:08:26.986  INFO teleinfo2mqtt: Connecting to port [/dev/ttyACM0] with history TIC mode
12:08:26.997  INFO teleinfo2mqtt: Connected to port [/dev/ttyACM0]
12:08:27.004  INFO teleinfo2mqtt: Connecting to MQTT broker [mqtt://core-mosquitto]
12:08:27.255  INFO teleinfo2mqtt: Connected to MQTT broker [mqtt://core-mosquitto]

**Reply by Goj34** [reply]
ça semble bien communiquer entre le broker et teleinfo.
quand je branche sur une prise USB (bonne idée ;) )  j'ai un très faible voyant rouge comme sur le raspberry.

**Reply by Goj34** [reply]
la connexion au broker:  2023-10-20 17:16:49: Client mqttjs_3edd6393 closed its connection.
2023-10-20 17:16:54: New connection from 172.30.33.4:43122 on port 1883.
2023-10-20 17:16:54: New client connected from 172.30.33.4:43122 as mqttjs_c9765bcc (p2, c1, k60, u'homeassistant').

**Reply by haade-administrator** [reply]
il y a un soucis avec la clé

**Comment by Makokoy79** [comment]
Salut, et d'abord merci pour ton travail.
Je suis en train d'essayer l'intégration de la clef de Charles sous Home assistant. 
J'ai d'abord testé la clef sous Linux, et j'obtiens les messages suivants :
               MSG1    PAS DE          MESSAGE                 <
               PRM     16168885471504  A
               RELAIS  000     B
               NTARF   02      O
               NJOURF  00      &
               NJOURF+1        00      B
               PJOURF+1        0000C001 08008002 NONUTILE NONUTILE NONUTILE NONUTILE NONUTILE NONUTILE NONUTILE                NONUTILE NONUTILE    (
               ADSC    041864289085    D
               VTIC    02      J
               DATE    H231103162702           =
               NGTF    H PLEINE/CREUSE         \
               LTARF     HEURE  PLEINE         A
               EAST    034387475       8
               EASF01  010291614       :
               EASF02  024095861       F
               EASF03  000000000       $
               EASF04  000000000       %
               EASF05  000000000       &
               EASF06  000000000       '
               EASF07  000000000       (
               EASF08  000000000       )
               EASF09  000000000       *
               EASF10  000000000       "
               EASD01  007299530       C
               EASD02  019555407       E
               EASD03  002992078       G
               EASD04  004540460       :
               IRMS1   005     3
               URMS1   239     H
               PREF    12      B
               PCOUP   12      \
               SINSTS  01130   K
               SMAXSN  H231103094426   04787   J
               SMAXSN-1        H231102092912   04665
               CCASN   H231103160000   01086   ;
               CCASN-1 H231103150000   01037   T
               UMOY1   H231103162000   238     .
               STGE    003AC401        Q
               MSG1    PAS DE          MESSAGE                 <

Je me dis que quelque chose passe, et que les messages sont cohérents (même si pour le moment je ne connais pas leur signification).

En tous cas l'intégrations sou HA ne se fait pas bien.

Côté **teleinfo2mqtt** le journal donne ceci (désolé c'est un peu long)
16:47:54.777  INFO teleinfo2mqtt: Connecting to port [/dev/serial/by-id/usb-CH2i_uTinfo-V3.0_TINFO-2397-if00] with standard TIC mode
16:47:54.788  INFO teleinfo2mqtt: Connected to port [/dev/serial/by-id/usb-CH2i_uTinfo-V3.0_TINFO-2397-if00]
16:47:54.796  INFO teleinfo2mqtt: Connecting to MQTT broker [mqtt://core-mosquitto:1883]
16:47:55.061  INFO teleinfo2mqtt: Connected to MQTT broker [mqtt://core-mosquitto:1883]
16:48:05.312  INFO teleinfo2mqtt: Publish configuration for tag ADSC for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_adsc/config]
16:48:05.320  INFO teleinfo2mqtt: Publish configuration for tag VTIC for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_vtic/config]
16:48:05.321  INFO teleinfo2mqtt: Publish configuration for tag DATE for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_date/config]
16:48:05.322  INFO teleinfo2mqtt: Publish configuration for tag NGTF for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_ngtf/config]
16:48:05.322  INFO teleinfo2mqtt: Publish configuration for tag LTARF for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_ltarf/config]
16:48:05.323  INFO teleinfo2mqtt: Publish configuration for tag EAST for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_east/config]
16:48:05.324  INFO teleinfo2mqtt: Publish configuration for tag EASF01 for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_easf01/config]
16:48:05.325  INFO teleinfo2mqtt: Publish configuration for tag EASF02 for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_easf02/config]
16:48:05.325  INFO teleinfo2mqtt: Publish configuration for tag EASF03 for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_easf03/config]
16:48:05.326  INFO teleinfo2mqtt: Publish configuration for tag EASF04 for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_easf04/config]
16:48:05.326  INFO teleinfo2mqtt: Publish configuration for tag EASF05 for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_easf05/config]
16:48:05.327  INFO teleinfo2mqtt: Publish configuration for tag EASF06 for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_easf06/config]
16:48:05.327  INFO teleinfo2mqtt: Publish configuration for tag EASF07 for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_easf07/config]
16:48:05.328  INFO teleinfo2mqtt: Publish configuration for tag EASF08 for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_easf08/config]
16:48:05.328  INFO teleinfo2mqtt: Publish configuration for tag EASF09 for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_easf09/config]
16:48:05.329  INFO teleinfo2mqtt: Publish configuration for tag EASF10 for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_easf10/config]
16:48:05.329  INFO teleinfo2mqtt: Publish configuration for tag EASD01 for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_easd01/config]
16:48:05.329  INFO teleinfo2mqtt: Publish configuration for tag EASD02 for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_easd02/config]
16:48:05.330  INFO teleinfo2mqtt: Publish configuration for tag EASD03 for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_easd03/config]
16:48:05.330  INFO teleinfo2mqtt: Publish configuration for tag EASD04 for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_easd04/config]
16:48:05.331  INFO teleinfo2mqtt: Publish configuration for tag IRMS1 for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_irms1/config]
16:48:05.331  INFO teleinfo2mqtt: Publish configuration for tag URMS1 for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_urms1/config]
16:48:05.331  INFO teleinfo2mqtt: Publish configuration for tag PREF for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_pref/config]
16:48:05.332  INFO teleinfo2mqtt: Publish configuration for tag PCOUP for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_pcoup/config]
16:48:05.332  INFO teleinfo2mqtt: Publish configuration for tag SINSTS for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_sinsts/config]
16:48:05.333  INFO teleinfo2mqtt: Publish configuration for tag SMAXSN for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_smaxsn/config]
16:48:05.333  INFO teleinfo2mqtt: Publish configuration for tag SMAXSN-1 for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_smaxsn-1/config]
16:48:05.334  INFO teleinfo2mqtt: Publish configuration for tag CCASN for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_ccasn/config]
16:48:05.334  INFO teleinfo2mqtt: Publish configuration for tag CCASN-1 for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_ccasn-1/config]
16:48:05.334  INFO teleinfo2mqtt: Publish configuration for tag UMOY1 for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_umoy1/config]
16:48:05.335  INFO teleinfo2mqtt: Publish configuration for tag STGE for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_stge/config]
16:48:05.335  INFO teleinfo2mqtt: Publish configuration for tag MSG1 for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_msg1/config]
16:48:05.335  INFO teleinfo2mqtt: Publish configuration for tag PRM for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_prm/config]
16:48:05.336  INFO teleinfo2mqtt: Publish configuration for tag RELAIS for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_relais/config]
16:48:05.336  INFO teleinfo2mqtt: Publish configuration for tag NTARF for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_ntarf/config]
16:48:05.337  INFO teleinfo2mqtt: Publish configuration for tag NJOURF for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_njourf/config]
16:48:05.337  INFO teleinfo2mqtt: Publish configuration for tag NJOURF+1 for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_njourf+1/config]
16:48:05.338  INFO teleinfo2mqtt: Publish configuration for tag PJOURF+1 for discovery to topic [homeassistant/sensor/teleinfo2mqtt/041864289085_pjourf+1/config]
16:48:05.409  WARN teleinfo2mqtt: Error when publishing to the mqtt broker (read ECONNRESET)
16:48:06.413  INFO teleinfo2mqtt: Reconnecting to the MQTT broker...

On voit une erreur à la fin, je ne sais pas pourquoi elle arrive. 

Enfin le journal de **Mosquitto broker** renvoie ceci :
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
cont-init: info: running /etc/cont-init.d/mosquitto.sh
[16:32:48] INFO: Setting up user mqtt
[16:32:49] INFO: SSL is not enabled
cont-init: info: /etc/cont-init.d/mosquitto.sh exited 0
cont-init: info: running /etc/cont-init.d/nginx.sh
cont-init: info: /etc/cont-init.d/nginx.sh exited 0
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
services-up: info: copying legacy longrun mosquitto (no readiness notification)
services-up: info: copying legacy longrun nginx (no readiness notification)
s6-rc: info: service legacy-services successfully started
[16:32:50] INFO: Starting NGINX for authentication handling...
[16:32:51] INFO: Starting mosquitto MQTT broker...
2023-11-03 16:32:51: Warning: Mosquitto should not be run as root/administrator.
2023-11-03 16:32:51: mosquitto version 2.0.17 starting
2023-11-03 16:32:51: Config loaded from /etc/mosquitto/mosquitto.conf.
2023-11-03 16:32:51: Loading plugin: /usr/share/mosquitto/go-auth.so
2023-11-03 16:32:51:  ├── Username/password checking enabled.
2023-11-03 16:32:51:  ├── TLS-PSK checking enabled.
2023-11-03 16:32:51:  └── Extended authentication not enabled.
2023-11-03 16:32:51: Opening ipv4 listen socket on port 1883.
2023-11-03 16:32:51: Opening ipv6 listen socket on port 1883.
2023-11-03 16:32:51: Opening websockets listen socket on port 1884.
2023-11-03 16:32:51: mosquitto version 2.0.17 running
2023-11-03 16:32:52: New connection from 127.0.0.1:37670 on port 1883.
2023-11-03 16:32:52: Client <unknown> disconnected due to protocol error.
[16:32:54] INFO: Successfully send discovery information to Home Assistant.
[16:32:56] INFO: Successfully send service information to the Supervisor.
2023-11-03 16:34:33: New connection from 172.30.33.7:57314 on port 1883.
2023-11-03 16:34:33: New client connected from 172.30.33.7:57314 as mqttjs_4fbc58ee (p2, c1, k60, u'mqtt').
2023-11-03 16:34:42: Client mqttjs_4fbc58ee disconnected due to malformed packet.
2023-11-03 16:34:44: New connection from 172.30.33.7:40992 on port 1883.
2023-11-03 16:34:44: New client connected from 172.30.33.7:40992 as mqttjs_4fbc58ee (p2, c1, k60, u'mqtt').
2023-11-03 16:44:22: Client mqttjs_4fbc58ee closed its connection.
2023-11-03 16:44:24: New connection from 172.30.33.7:55910 on port 1883.
2023-11-03 16:44:24: New client connected from 172.30.33.7:55910 as mqttjs_a1e8e85e (p2, c1, k60, u'mqtt').
2023-11-03 16:44:35: Client mqttjs_a1e8e85e disconnected due to malformed packet.
2023-11-03 16:44:36: New connection from 172.30.33.7:52062 on port 1883.
2023-11-03 16:44:36: New client connected from 172.30.33.7:52062 as mqttjs_a1e8e85e (p2, c1, k60, u'mqtt').
2023-11-03 16:44:56: Client mqttjs_a1e8e85e closed its connection.
2023-11-03 16:44:59: New connection from 172.30.33.7:38402 on port 1883.
2023-11-03 16:44:59: New client connected from 172.30.33.7:38402 as mqttjs_e06bb322 (p2, c1, k60, u'mqtt').
2023-11-03 16:45:17: Client mqttjs_e06bb322 closed its connection.
2023-11-03 16:45:20: New connection from 172.30.33.7:41824 on port 1883.
2023-11-03 16:45:20: New client connected from 172.30.33.7:41824 as mqttjs_4be24334 (p2, c1, k60, u'mqtt').
2023-11-03 16:47:47: Client mqttjs_4be24334 closed its connection.
2023-11-03 16:47:49: New connection from 172.30.33.7:38782 on port 1883.
2023-11-03 16:47:49: New client connected from 172.30.33.7:38782 as mqttjs_1b5ff69f (p2, c1, k60, u'mqtt').
2023-11-03 16:47:52: Client mqttjs_1b5ff69f closed its connection.
2023-11-03 16:47:55: New connection from 172.30.33.7:39190 on port 1883.
2023-11-03 16:47:55: New client connected from 172.30.33.7:39190 as mqttjs_23e32595 (p2, c1, k60, u'mqtt').
2023-11-03 16:48:05: Client mqttjs_23e32595 disconnected due to malformed packet.
2023-11-03 16:48:06: New connection from 172.30.33.7:48144 on port 1883.
2023-11-03 16:48:06: New client connected from 172.30.33.7:48144 as mqttjs_23e32595 (p2, c1, k60, u'mqtt').
2023-11-03 17:02:53: Saving in-memory database to /data//mosquitto.db.

Beaucoup de connexions et déconnexions.

Désolé je te bombarde de lignes, je ne sais pas ce qui est important de ce qui ne l'est pas... Je bataille depuis ce milieu d'après midi sans trop savoir où aller. Sais tu m'aiguiller sur mon problème ?

Merci d'avance !

**Reply by Makokoy79** [reply]
Je viens de regarder avec le document de ENEDIS disponible ici : https://www.enedis.fr/media/2035/download
Les mots que je reçois sous Linux sont cohérents de l'attendu.

**Reply by haade-administrator** [reply]
fait remonter le problème similaire sur le github de fmartinou:
[https://github.com/fmartinou/teleinfo2mqtt/issues/60](https://github.com/fmartinou/teleinfo2mqtt/issues/60)

**Reply by Goj34** [reply]
Bonsoir @Makokoy79 .  on dirait un pb entre ton teleinfo2mqtt et ton broker MQTT. peut être un pb de configuration: 2023-11-03 16:32:52: Client disconnected due to protocol error.
il faudrait regarder les configurations des deux: MQTT et Teleinfo2mqtt.

A dispo
Goul

**Comment by BaptisteBruneau** [comment]
Bonjour 😊
Merci beaucoup pour l'article très détaillé ! 
Je suis assez novice, surtout tout ce qui est câblage 😁
Je me demande quel type de fil je peux utiliser du module vers le compteur Linky? 
Est ce que des câble électrique (rouge / bleu) ça fait l'affaire ? Ou bien faut il prévoir autre chose ? 
Merci par avance pour un éventuel retour 😁

**Comment by djedje6821** [comment]
Salut et merci pour ce beau tuto ! C'est toujours plaisant quand des gens passent du temps pour documenter dans le but d'aider les autres... 
J'ai une petite question, je suis peut être passé à côté, mais j'ai la Téléinfo v2 qui est branché sur un Pi Zero directement connecté sur mon Linky. Du coup mon HAOS est installé sur une autre VM, et je ne trouve pas vraiment comment configuré l'addon teleinfo2mqtt add-on dans ce type d'utilisation... Est-ce que tu pourrais m'aiguiller stp ?

**Reply by djedje6821** [reply]
Je pense que dois installer teleinfo2mqtt en Docker sur mon Pi Zero ?

**Comment by ClasheGit** [comment]
Bonjour, 
J'ai en ma possession une interface micromodule téléinfo V3 et un raspberry, mon HA est sur une VM mais trop éloigné du compteur électrique.
Je n'arrive pas à comprendre ce que vous installez sur le raspberry pour faire fonctionner la micro module, depuis HA j'ai bien installer ce qu'il faut il attend maintenant l'adresse du raspberry configuré.

**Comment by philzilla** [comment]
Bonjour,

Merci beaucoup pour ton tuto, tout fonctionne !
Pour info j'ai utilisé le Téléinfo v2

**Comment by Hugo158415** [comment]
Bonjour,
je crois que j'ai fait 10 fois la manip et j'ai toujours le même message d'erreur: Module complémentaire introuvable
ça fait une semaine que j'ai acheté ce module et vraiment je regrette mon HA
